google_analytics/aggregation.py

```diff
@@ -4,10 +4,32 @@
 import pandas as pd
 
 
+# Specify what to do with each column
+# Static
+OHE = ['channelGrouping', 'browser', 'deviceCategory', 'operatingSystem', 'city', 'continent',
+       'country', 'metro', 'region', 'subContinent', 'adContent',
+       'adwordsClickInfo.adNetworkType', 'adwordsClickInfo.page', 'adwordsClickInfo.slot',
+       'campaign', 'medium', 'source_cat', 'weekday', 'visitHour']
+OHE_reduced = ['channelGrouping', 'browser', 'deviceCategory', 'operatingSystem', 'city', 'continent',
+               'country', 'metro', 'region', 'subContinent', 'adContent',
+               'adwordsClickInfo.adNetworkType', 'adwordsClickInfo.page', 'adwordsClickInfo.slot',
+               'campaign', 'medium']  # the categories to be reduced
+booleans = ['isMobile', 'adwordsClickInfo.isVideoAd', 'isTrueDirect', 'keyword.isGoogle', 'keyword.isYouTube']
+cat_nunique = ['networkDomain']
+num_mean = ['totalVisits', 'keyword.mistakes_Google', 'keyword.mistakes_YouTube']
+
+# Dynamic
+monthly_count = ['bounces', 'newVisits']
+monthly_mean = ['hits', 'pageviews', 'target']
+monthly_sum = ['hits', 'pageviews', 'target']
+
+
 def load_train_test_dataframes(data_dir, x_train_file_name='preprocessed_x_train.csv',
                                y_train_file_name='preprocessed_y_train.csv',
```

Comment:
we don't need both `data_dir` and the filenames. Just accept `train_path = <data_dir>/filename`. Also we only need to accept the argument once.